src/librustc/mir/interpret/allocation.rs

```diff
@@ -106,13 +109,33 @@ impl<Tag> Allocation<Tag> {
             bytes: vec![0; size.bytes() as usize],
             relocations: Relocations::new(),
             undef_mask: UndefMask::new(size, false),
+            size,
             align,
             mutability: Mutability::Mutable,
             extra: (),
         }
     }
 }
 
+/// Raw accessors. Provide access to otherwise private bytes.
+impl<Tag, Extra> Allocation<Tag, Extra> {
+    pub fn len(&self) -> usize {
+        self.size.bytes() as usize
+    }
+
+    /// Look at a slice which may describe undefined bytes or describe a relocation. This differs
+    /// from `get_bytes_with_undef_and_ptr` in that it does no bounds checks and relocation checks
```

Comment:
`get_bytes_with_undef_and_ptr` also does not do bounds checks.

Reply:
Also please extend the doc comment to say something like "every use of this method should come with justification for why `get_bytes_with_undef_and_ptr` wasn't used".

Reply:
Probably it should also say that this should only be used when inspecting the interpreter state "from the outside", not for interpreter execution. Maybe that can be reflected in the name?

Reply:
Maybe `dump_bytes_with_undef_and_ptr` or `uninterpreted_bytes_with_undef_and_ptr`?

Reply:
`dump` sounds like a debugging thinking dump to stderr.

`uninterpreted` doesn't really convey anything `raw` doesn't convey. `meta_level_inspect_bytes_with_undef_and_ptr` would work for me but I doubt it works for many others here.^^

Reply:
I'm all for verbose names when that definitely clarifies it. Better than having a method name and misinterpreting what it means because no one reads docs

Reply:
> because no one reads docs

@oli-obk  I will have you that I definitely read docs (with a convenient firefox key `rustc`) when doing reviews where I haven't seen a method. =)

Reply:
@RalfJung You're right, those aren't much better. I would find `meta` to be too generic, my first thought would be that it returns a property *of the bytes* rather than returning a raw part of the interpreter state. It works somewhat similar to a debugger attached to the interpreter, doesn't it? Maybe `inspect_internal_raw_bytes_with_undef_and_ptr` makes it clear that this is not to be used as part of interpreter execution.

Reply:
I was thinking `inspect` as well (not sure `internal` adds anything), but one could reasonably say that the machine also "inspects" its bytes.

`inspect_from_outside` or so... maybe?^^

Reply:
Combining parts of the above: `inspect_bytes_with_undef_and_ptr_outside_interpreter`?

Reply:
Well, that's really long. But it does convey the crucial point.

Reply:
Maybe the `_bytes`-part could be dropped. 'inspect' and 'with_undef_and_ptr' already make it clear which part of the allocation is returned. That drops it to slightly below half the line length.

Reply:
So that would be `inspect_with_undef_and_ptr_outside_interpreter`. @oli-obk is that excessive?

Reply:
No, not excessive, it makes it very clear that something nontrivial is going on and gives a good hint as to what that thing is.